File: app/lib/models/order/order_item.dart
import 'package:json_annotation/json_annotation.dart';

part '../../generated/order/order_item.g.dart';

@JsonSerializable()
class OrderItem {
  final String? id;
  final String cartItemId;
  final String productId;
  final String productVariantId;
  final String imageUrl;
  final String productName;
  final String variantName;
  final int quantity;
  final double price;

  OrderItem({
    this.id,
    required this.cartItemId,
    required this.productId,
    required this.productVariantId,
    required this.imageUrl,
    required this.productName,
    required this.variantName,
    required this.quantity,
    required this.price,
  });

  factory OrderItem.fromJson(Map<String, dynamic> json) => _$OrderItemFromJson(json);

  Map<String, dynamic> toJson() => _$OrderItemToJson(this);
}

File: api/src/main/java/com/project/supershop/features/orders/domain/entities/OrderItem.java
package com.project.supershop.features.orders.domain.entities;

import com.project.supershop.common.BaseEntity;
import com.project.supershop.features.orders.domain.dto.requests.OrderItemRequest;
import com.project.supershop.features.product.domain.entities.Product;
import jakarta.persistence.Entity;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.SuperBuilder;

import java.time.LocalDateTime;

@Entity
@Table(name = "orderItem")
@NoArgsConstructor
@AllArgsConstructor
@Data
@SuperBuilder
public class OrderItem extends BaseEntity {
    private String cartItemId;
    private String imageUrl;
    private String productName;
    private String variantName;
    private String productId;
    private String productVariantId;
    private Integer quantity;
    private Integer price;

    @ManyToOne
    @JoinColumn(name = "orderId")
    private Order order;

    public static OrderItem createOrderItem(OrderItemRequest orderItemRequest, Order order){
        return OrderItem.builder()
                .order(order)
                .cartItemId(orderItemRequest.getCartItemId())
                .imageUrl(orderItemRequest.getImageUrl())
                .productName(orderItemRequest.getProductName())
                .variantName(orderItemRequest.getVariantName())
                .productId(orderItemRequest.getProductId())
                .productVariantId(orderItemRequest.getProductVariantId())
                .quantity(orderItemRequest.getQuantity())
                .price(orderItemRequest.getPrice())
                .build();
    }
}
